### PR TITLE
Redis / Docker Compose

### PR DESCRIPTION
**Pull request by jkassemi**
* Adding docker and docker-compose files

This commit adds in a dockerfile to build a container for the vets-api and a docker-compose file to bring up a full stack for development or testing purposes

* Migrate docker-compose to rails-config settings

* pdftk, clamav and imagemagick configuration for container

### CODE REVIEW

**Comment by patrickvinograd** [comment]
Gonna verify the instructions...

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+BASH_TEST		:= $(COMPOSE_TEST) run vets-api bash --login -c
+
+.PHONY: default
+default: ci
+
+.PHONY: ci
+ci:
+	@$(BASH_TEST) "bundle exec rake db:setup db:migrate ci"
+
+.PHONY: console
```
Comment (patrickvinograd) [inline Makefile]: I'm not convinced `console` should depend on `db` - at least in my typical workflow when I want to use rails console I don't want to run migrations.

File: README.md [hunk 2]
```diff
@@ -2,68 +2,25 @@
...
-Vets API requires:
-- PostgreSQL
-- Redis
-- Ruby 2.3
-
 ### Base Setup
 
 To start, fetch this code:
 
 `git clone https://github.com/department-of-veterans-affairs/vets-api.git`
```
Comment (patrickvinograd) [inline README.md]: I think we should check with all developers before we make docker the only documented mechanism for running vets-api. I know there were some who didn't have easy root/sudo access on various machines. I don't think anything you've done makes it impossible to use the old steps so maybe we document Docker as the preferred approach but leave the old instructions, possibly in a separate file for clarity.

File: README.md [hunk 3]
```diff
@@ -2,68 +2,25 @@
...
-   - `touch ~/.certs/vetsgov-localhost.crt`
-   - `touch ~/.certs/vetsgov-localhost.key`
-1. Create dev database: `bundle exec rake db:setup`
+     commands will provide basic functionality:
+   - `touch config/certs/vetsgov-localhost.crt`
+   - `touch config/certs/vetsgov-localhost.key`
+1. Run the vets-api dependencies and application
+    - `make up`
 
+The API will then be available on port 3000 of the docker host.
```
Comment (patrickvinograd) [inline README.md]: Maybe add a bit of context around Docker for those who aren't familiar. E.g if I change code do I need to run `make up` again, or are changes automatically reflected in the container? It's not obvious to everybody.

**Comment by patrickvinograd** [comment]
Documented steps work great, just want to make sure we can commit to that as the preferred approach for everybody.

**Comment by jkassemi** [comment]
@patrickvinograd can you give the changes a once-over?

**Pull request merged**